doc/high-level.md

```diff
@@ -0,0 +1,83 @@
+| `(from-list '(1d0 2d0 3d0 4d0 5d0 6d0) '(2 3))` | `[ 1 2 3; 4 5 6 ]` | `array([[1.,2.,3.], [4.,5.,6.]])` | Create a 2x3 matrix from given elements.                                             |
+| `(empty '(2 3 4))` or `(const 0d0 '(2 3 4))`    | `zeroes(2,3,4)`    | `zeroes((2,3,4))`                 | Create a 2x3x4 dimensional array of zeroes of double-float element type.             |
+| `(const 1d0 '(3 4))`                            | `ones(3,4)`        | `ones((3,4))`                     | Create a 3x4 dimensional array of ones of double-float element type.                 |
+| `(deye 1d0 '(3 3))`                             | `eye(3)`           | `eye(3)`                          | Create a 3x3 identity array of double-float element type.                            |
+| `(from-diag a (list (length a) (length a)))`    | `diag(a,0)`        | `diag(a,0)`                       | Create a square matrix from the diagonal entries in `a` with zeroes everywhere else. |
+| `(rand '(3 4))`                                 | `rand(3,4)`        | `random.rand(3,4)`                | Create a random 3x4 array.                                                           |
+
+### Basic Operations
+
+| MAGICL    | MATLAB   | NumPy            | Description                 |
+|-----------|----------|------------------|-----------------------------|
+| `(@ a b)` | `a * b`  | `a @ b`          | Matrix multiplication       |
+| `(+ a b)` | `a + b`  | `add(a, b)`      | Element-wise add            |
+| `(- a b)` | `a - b`  | `subtract(a, b)` | Element-wise subtract       |
+|           | `a .* b` | `a * b`          | Element-wise multiply       |
+|           | `a./b`   | `a/b`            | Element-wise divide         |
+|           | `a.^3`   | `a**3`           | Element-wise exponentiation |
+
+### Linear Algebra
+
+| MAGICL                                                  | MATLAB            | NumPy                                                          | Description                                |
+|---------------------------------------------------------|-------------------|----------------------------------------------------------------|--------------------------------------------|
+| `(det a)`                                               | `det(a)`          | `linalg.det(a)`                                                | Determinant of matrix                      |
+| `(trace a)`                                             | `trace(a)`        | `trace(a)`                                                     | Trace (sum of diagonal elements) of matrix |
+| `(upper-triangular a)`                                  | `triu(a)`         | `triu(a)`                                                      | Upper triangular part of matrix            |
+| `(lower-triangular a)`                                  | `tril(a)`         | `tril(a)`                                                      | Lower triangular part of matrix            |
```

Comment:
I'd at least provide synonyms of these with `triu` and `tril`. The current names are neat for programming but are really long and annoying to type when you're in the REPL prototyping something.

Reply:
Added in 39e68ed